base/ocsp/src/main/java/org/dogtagpki/server/ocsp/OCSPEngine.java

```diff
@@ -149,9 +150,13 @@ public boolean isRevoked(X509Certificate[] certificates) {
         }
 
         for (X509Certificate cert: certificates) {
+            // validateConnCertWithCRL only handles leaf certs
+            if (CertUtils.isCACert(cert))
+                continue;
```

Comment:
btw, I didn't realize there were comments before I clicked merge, so I'm replying after the fact.  sorry about it.  So, to answer your questions. The crl cert validation is only for validating peer certs.  1. in the pki setup, we never use a ca or self-signed cert for SSL connections, either a client cert, or a server cert.  2. yes all non-leaf certs are supposed to be ca certs, hence forming a chain. 3. Yes your proposal of using CertificateChain to sort is a very good idea.  I actually copied a part of this code from JssSubsystem isCACert().  How about we create a separate ticket to combine the two isCACert() and make the improvement at the same time?  I'll leave it as it for now, since Pritam has already tested it. Thanks!